src/config/app.ts

```diff
@@ -0,0 +1,353 @@
+import fs, {PathLike} from 'fs';
+import semver from 'semver';
+import parseJsonFile from '../json';
+import {Tool, ToolExecutionType} from '../tools';
+import {CURRENT_STABLE, INSTALLABLE_VERSIONS, InstallablePhpVersionType, isInstallableVersion} from './php';
+import {ComposerJson} from './composer';
+import {ConfigurationFromFile, isAdditionalChecksConfiguration, isAnyComposerDependencySet, isAnyPhpVersionType, isConfigurationContainingJobExclusions, isExplicitChecksConfiguration, isLatestPhpVersionType, isLowestPhpVersionType, JobDefinitionFromFile, JobFromFile, JobToExcludeFromFile, WILDCARD_ALIAS} from './input';
+
+export const OPERATING_SYSTEM = 'ubuntu-latest';
+export const ACTION = 'laminas/laminas-continuous-integration-action@v1';
+
+export enum ComposerDependencySet {
+    LOWEST = 'lowest',
+    LOCKED = 'locked',
+    LATEST = 'latest',
+}
+
+function gatherVersions(composerJson: ComposerJson): InstallablePhpVersionType[] {
+    if (JSON.stringify(composerJson) === '{}') {
+        return [];
+    }
+
+    const composerPhpVersion: string = (composerJson.require?.php ?? '').replace(/,\s/, ' ');
+
+    if (composerPhpVersion === '') {
+        return [];
+    }
+
+    return INSTALLABLE_VERSIONS
+        .filter((version) => semver.satisfies(`${version}.0`, composerPhpVersion));
```

Comment:
Hmm, probably inherent from original design, but this means we need to touch this action at each release 🤔 

Non-issue for porting to TS, just a problem for the future.